dirty_cat/test/test_super_vectorizer.py

```diff
@@ -18,90 +19,171 @@ def check_same_transformers(expected_transformers: dict, actual_transformers: li
     assert actual_transformers_dict == expected_transformers
 
 
-def _get_dataframe():
+def _get_clean_dataframe():
+    """
+    Creates a simple DataFrame without missing values.
+    """
     return pd.DataFrame({
-        'int': pd.Series([15, 56, 63, 12, 44], dtype=int),
-        'float': pd.Series([5.2, 2.4, 6.2, 10.45, 9.], dtype=float),
+        'int': pd.Series([15, 56, 63, 12, 44], dtype='int'),
+        'float': pd.Series([5.2, 2.4, 6.2, 10.45, 9.], dtype='float'),
         'str1': pd.Series(['public', 'private', 'private', 'private', 'public'], dtype='string'),
         'str2': pd.Series(['officer', 'manager', 'lawyer', 'chef', 'teacher'], dtype='string'),
         'cat1': pd.Series(['yes', 'yes', 'no', 'yes', 'no'], dtype='category'),
         'cat2': pd.Series(['20K+', '40K+', '60K+', '30K+', '50K+'], dtype='category'),
     })
 
 
-def test_super_vectorizer():
-    # Create a simple DataFrame
-    X = _get_dataframe()
+def _get_dirty_dataframe():
+    """
+    Creates a simple DataFrame with missing values.
+    """
+    return pd.DataFrame({
+        'int': pd.Series([15, 56, 63, 12, 44], dtype='int'),
+        'float': pd.Series([5.2, 2.4, 6.2, 10.45, 9.], dtype='float'),
+        'str1': pd.Series(['public', np.nan, 'private', 'private', 'public'], dtype='object'),
+        'str2': pd.Series(['officer', 'manager', np.nan, 'chef', 'teacher'], dtype='object'),
+        'cat1': pd.Series([np.nan, 'yes', 'no', 'yes', 'no'], dtype='object'),
+        'cat2': pd.Series(['20K+', '40K+', '60K+', '30K+', np.nan], dtype='object'),
```

Comment:
For recent pandas version, we should also test with Na on ints
 https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html#integer-dtypes-and-missing-data